### ISSUE TITLE
Unable to get HIPoint object of the tapped point using HIContext !!??

### ISSUE THREAD

**Issue by SSJ2912** [highcharts/highcharts-android#60]
I am looking to replicate the following JS code in Android using Kotlin and I am unable to get any HighCharts object using the HIContext object provided inside the HIFunction !

JS Code ->

```
 point:{
                  events:{
                    click: function(){
                      moveToPoint(this);
                    }
                  }
                }

 function moveToPoint(clickPoint){
      var points = clickPoint.series.points;
      var startAngle = 0;
      for (var i = 0; i < points.length; i++){
        var p = points[i];
        if (p == clickPoint)
        {
          break;
        }
        startAngle += (p.percentage/100.0 * 360.0);
      }
      clickPoint.series.update({
        //startAngle: -startAngle + 180 // start at 180
        startAngle: -startAngle + 180 - ((clickPoint.percentage/100.0 * 360.0)/2) // center at 180
      });
    }
```

Please guide me  on how one would achieve this in Android !
I have also attached my Kotlin code below,

    chartOption.plotOptions.series.point.events.click = HIFunction(   HIConsumer<HIChartContext> {
            val point = it.getProperty("this.point") as HIPoint
            val series = it.getProperty("this.series") as HISeries
        } , arrayOf("this.point","this.series"))

**Comment by soommy12** [comment]
Hello @SSJ2912 ! Thanks for the question. In the array of arguments for your `HIFunction` you shouldn't use `this` because the `HIChartView` contains only one chart instance. Passing `point` and `series` will be enough.

**Comment by SSJ2912** [comment]
Hi @soommy12 , Passing `point` or `series` is also not working. It throws an uncaught exception "Uncaught TypeError: Converting circular structure to JSON"

The code :
```
options.plotOptions.series.point.events.click = HIFunction(   HIConsumer<HIChartContext> {
                val y = it.getProperty("y")
                val series = it.getProperty("series") as HISeries
            } , arrayOf("y" , "series" ))
```

**Comment by soommy12** [comment]
Yes, that is right @SSJ2912 . 

**TL;DR:** 
Some of the Highcharts objects are more complex than others and you can't get them from chart click event. Keep in mind that the library is wrapping the core!

**Long explanation:**
In other words: the Java/Kotlin languages have some limitations when it comes to converting chart specific elements back to the Android environment. When you're trying to pull something from `HIChartContext` those objects are not defined in your code any more. The library is trying it's best to convert them back to Java/Kotlin object but some more complicated fields (like series) won't make it's way back.
